### PR TITLE
Preserve the case of username and password in URL.build

### PR DESCRIPTION
**Pull request by tushar-balwani**
## What do these changes do?
The new changes preserve casing of username and password when extracted from host in URL.build

```python
>>> yarl.URL.build(scheme='httPS', host='usER:passWORD@hostNAME', path='/paTH')
URL('httPS://usER:passWORD@hostname/paTH')
>>> yarl.URL.build(scheme='httPS', host='usER:passWORD@hostNAME', path='/paTH').user
'usER'
>>> yarl.URL.build(scheme='httPS', host='usER:passWORD@hostNAME', path='/paTH').password
'passWORD'
```

## Are there changes in behavior for the user?

No.

## Related issue number

#880 

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: yarl/_url.py [hunk 1]
```diff
@@ -261,6 +261,12 @@ def build(
         elif not user and not password and not host and not port:
             netloc = ""
         else:
+            if "@" in host and not user and not password:
```
Comment (webknjaz) [inline yarl/_url.py:264]: This seems like a weird place to extract this data from.. Could you add more detail regarding the problem you're trying to address?

File: tests/test_url_build.py [hunk 2]
```diff
@@ -257,3 +257,14 @@ def test_build_with_none_query_string():
...
+
+def test_build_with_case_sensitive_user_and_password():
+    url = URL.build(scheme='httPS', host='usER:passWORD@hostNAME', path='/paTH')
+    assert url.user == "usER"
+    assert url.password == "passWORD"
+
+
+def test_build_with_case_sensitive_user():
+    url = URL.build(scheme='httPS', host='usER@hostNAME', path='/paTH')
+    assert url.user == "usER"
```
Comment (webknjaz) [inline tests/test_url_build.py:270]: Looks like these are the same test. So I'd recommend using `@pytest.mark.parametrize` to fold it.

Also, I don't believe that scheme needs to be case-insensitive. Do you have any RFC reference that clarify this corner case?

File: CHANGES/880.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Preserve the case of username and password in URL.build when extracted from host.
```
Comment (webknjaz) [inline CHANGES/880.bugfix.rst:1]: Please, use the past tense and RST markup here.

File: tests/test_url_build.py [hunk 4]
```diff
@@ -257,3 +257,14 @@ def test_build_with_none_query_string():
 def test_build_with_none_fragment():
     with pytest.raises(TypeError):
         URL.build(scheme="http", host="example.com", fragment=None)
+
+
+def test_build_with_case_sensitive_user_and_password():
+    url = URL.build(scheme='httPS', host='usER:passWORD@hostNAME', path='/paTH')
```
Comment (webknjaz) [inline tests/test_url_build.py:263]: Also, why this information would ever be a part of the host? I don't think the API should allow for it at all...

**Comment by mjpieters** [comment]
This is the wrong fix, as discussed in #880. The issue is that `host` should only accept a host string, not an _authority_ string. Pass those to `URL.build(authority=...)` instead.

The right fix is to reject any [`gen-delims` characters](https://www.rfc-editor.org/rfc/rfc3986#section-2.2) in the host value (provided it is not an IP address). Either by raising a deprecation warning first, or just going straight to raising `ValueError`.
